router/src/db.rs

```diff
@@ -112,18 +112,12 @@ impl Db {
         state.entries.insert(id, entry);
     }
 
-    /// Remove an entry from the database if it exists
-    pub(crate) fn remove(&self, id: &u64) -> Option<Entry> {
-        let mut state = self.shared.state.lock();
-        state.entries.remove(id)
-    }
-
     // Get the next batch
     pub(crate) fn next_batch(
         &self,
         min_size: Option<usize>,
         max_size: usize,
-    ) -> Option<(Vec<u64>, Batch)> {
+    ) -> Option<(HashMap<u64, Entry>, Batch)> {
```

Comment:
I guess this ends up slightly cleaner code-wise than adding to an existing hashmap as is done my PR, but it means a bit more "happy path" work is done when extending a batch ... i.e. populating a new map and then transferring its contents to another one rather than just inserting the new entries once.

Reply:
Here's a [ref](https://nnethercote.github.io/perf-book/heap-allocations.html#reusing-collections) for this one.

Reply:
I preffer the api like this. I ll keep this in my mind though if we need to squeeze performance later.